osf/migrations/0163_populate_file_versions_through.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-03-03 17:52
+from __future__ import unicode_literals
+
+import logging
+from math import ceil
+
+from django.db import migrations, connection
+
+logger = logging.getLogger(__file__)
+
+increment = 10000
+
+def restore_default_through_table(state, schema):
+    sql = """
+
+        INSERT INTO osf_basefilenode_versions (basefilenode_id, fileversion_id)
+        SELECT new_thru.basefilenode_id, new_thru.fileversion_id
+        FROM osf_basefileversionsthrough new_thru;
+    """
+    with connection.cursor() as cursor:
+        cursor.execute(sql)
+
+
+# Batching adapted from strategy in website/search_migration/migrate.py
+def populate_fileversion_name(state, schema):
+    BaseFileNodeVersion = state.get_model('osf.basefilenode_versions')
+    max_thru_id = getattr(BaseFileNodeVersion.objects.order_by('id').last(), 'id', 0)
+
+    sql = """
+        INSERT INTO osf_basefileversionsthrough (basefilenode_id, fileversion_id, version_name)
+        SELECT F.id, V.id, F.name
+        FROM osf_basefilenode_versions THRU, osf_fileversion V, osf_basefilenode F
+        WHERE THRU.fileversion_id = V.id
+        AND THRU.basefilenode_id = F.id
+        AND THRU.id > {}
+        AND THRU.id <= {};
```

Comment:
yikes - this is going to take way too long for a migration, this table is huge.  I batched it into increments of 100,000 - it's on track to take over an hour.  Maybe this is a good candidate for the management command strategy?

Step 1:
Create new through table, but continue to write to both the old through table and the new through table.
Meanwhile, have a management command populate the new through table over time.

Step 2:
Switch the filenode.versions field to point to the new through table